yarn-project/pxe/src/kernel_prover/kernel_prover.ts

```diff
@@ -118,7 +121,12 @@ export class KernelProver {
         currentExecution.publicInputs.callContext.functionSelector,
       );
 
-      const appVk = await this.proofCreator.computeAppCircuitVerificationKey(currentExecution.acir, functionName);
+      // TODO (#6185): How do we know if this one should be recursive? The first is the application which comes from the chain (simulateTx).
+      const appVk = await this.proofCreator.computeAppCircuitVerificationKey(
+        currentExecution.acir,
+        recursive,
+        functionName,
+      );
```

Comment:
App circuits are always recursive see the `#[recursive]` attribute being applied [here](https://github.com/AztecProtocol/aztec-packages/blob/5f386963b06752087c2600949cbb4bb2910b25ef/noir-projects/aztec-nr/aztec/src/macros/functions/mod.nr#L216). I would expect the kernel circuits to also be recursive as well.

Reply:
Ah, thanks for that link, never would have found it on my own. I wonder why those still compile, since the `.nr` that has `#[recursive]` are rejected now.

I thought the kernel circuits are not recursive only because the ones I had to remove `#[recursive]` from are precisely the `ServerCircuitArtifact` kinds. For example `private-kernel-reset/src/main.nr` and `private-kernel-init/src/main.nr` for example don't have the attribute.

Reply:
Yeah, I'm not sure what's up with the private kernel circuits. I've been digging into how they're used a little more and it's likely that the client IVC proving scheme was ignoring this flag.

Reply:
Will need to do some more debugging

Reply:
Hm, but then the IVC proving function needs to change as well so that it treats the first ACIR as recursive and the rest of them as potentially not. I was wondering about whether you need two flags for that (first and last), or a full array of flags.

Reply:
You're right, the `ClientIVC` always called [add_recursive_proof](https://github.com/AztecProtocol/aztec-packages/blob/fd974e1ba91e01910751ed87da6dbeb068faba4f/barretenberg/cpp/src/barretenberg/client_ivc/client_ivc.cpp#L171)

Reply:
Ah yes, I'm fairly sure that the client IVC scheme is never meant to be verified onchain so it'll always be recursive. It gets fed into the "tube circuit" which spits out an ultrahonk proof.

Reply:
I can see a transaction being proven and sent to the aztec network [here](https://github.com/AztecProtocol/aztec-packages/blob/fd974e1ba91e01910751ed87da6dbeb068faba4f/yarn-project/aztec.js/src/contract/base_contract_interaction.ts#L78-L79) but I'm not sure that means it's going to be directly verified on-chain, or aggregated further. At least at that point the proving result does contain a `ClientIvcProof`. 

When is the tube circuit fed with it?

Reply:
I'm not sure. Let's pull in someone with more experience with how the transaction flow of aztec network translates into proofs tomorrow.

Reply:
I think you're right, it seems that as soon as the orchestrator gets the IVC proof in [addNewTx](https://github.com/AztecProtocol/aztec-packages/blob/fd974e1ba91e01910751ed87da6dbeb068faba4f/yarn-project/prover-client/src/orchestrator/orchestrator.ts#L245-L269) it ends up enqueuing it to be proven with tube.